core/src/main/java/org/elasticsearch/cluster/routing/OperationRouting.java

```diff
@@ -177,10 +178,17 @@ private ShardIterator preferenceActiveShardIterator(IndexShardRoutingTable index
             }
         }
         // if not, then use it as the index
+        int routingHash = Murmur3HashFunction.hash(preference);
+        if (nodes.getMinNodeVersion().onOrAfter(Version.V_6_0_0_alpha1_UNRELEASED)) {
+            // Better overall balancing can be achieved if each shardId opts
```

Comment:
can we say that the shard ordering is the determined by a fixed node order (and point to the AllocationService for details) and thus we need to make sure we scatter across shards? it's better to say why than just saying better ;)